admin/class-my-yoast-route.php

```diff
@@ -0,0 +1,198 @@
+<?php
+/**
+ * WPSEO plugin file.
+ *
+ * @package WPSEO\Admin
+ */
+
+/**
+ * Represents the route for MyYoast.
+ */
+class WPSEO_MyYoast_Route implements WPSEO_WordPress_Integration {
+
+	/**
+	 * @var string
+	 */
+	const PAGE_IDENTIFIER = 'wpseo_myyoast';
+
+	/**
+	 * Sets the hooks when the user has enough rights and is on the right page.
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+		$route = filter_input( INPUT_GET, 'page' );
+		if ( ! ( $this->is_myyoast_route( $route ) && $this->can_access_route() ) ) {
+			return;
+		}
+
+		if ( ! $this->is_valid_action( $this->get_action() ) ) {
+			return;
+		}
+
+		add_action( 'admin_menu', array( $this, 'register_route' ) );
+		add_action( 'admin_init', array( $this, 'handle_route' ) );
+	}
+
+	/**
+	 * Registers the page for the MyYoast route.
+	 *
+	 * @codeCoverageIgnore
+	 *
+	 * @return void
+	 */
+	public function register_route() {
+		add_dashboard_page(
+			'',
+			'',
+			'wpseo_manage_options',
+			self::PAGE_IDENTIFIER
+		);
+	}
+
+	/**
+	 * Abstracts the action from the url and follows the appropriate route.
+	 *
+	 * @return void
+	 */
+	public function handle_route() {
+		$action = $this->get_action();
+		switch ( $action ) {
+			case 'connect';
+				$this->connect();
+			break;
+		}
+
+		$this->stop_execution();
+	}
+
+	/**
+	 * Checks if the current page is the MyYoast route.
+	 *
+	 * @param string $route The myyoast route.
+	 *
+	 * @return bool True when url is the myyoast route.
+	 */
+	protected function is_myyoast_route( $route ) {
+		return ( $route === self::PAGE_IDENTIFIER );
+	}
+
+	/**
+	 * Compares an action to a list of allowed actions to see if it is valid.
+	 *
+	 * @param string $action The action to check.
+	 *
+	 * @return bool True if the action is valid.
+	 **/
+	protected function is_valid_action( $action ) {
+		$allowed_actions = array( 'connect' );
+
+		return in_array( $action, $allowed_actions );
+	}
+
+	/**
+	 * Connects to MyYoast, generates a ClientId if needed.
+	 *
+	 * @return void
+	 */
+	protected function connect() {
+		$config    = $this->get_client()->get_configuration();
+		$client_id = $config['clientId'];
+
+		if ( empty( $client_id ) ) {
+			$client_id = $this->generate_userid();
+
+			$this->get_client()->save_configuration(
+				array(
+					'clientId' => $client_id,
+				)
+			);
+		}
+
+		$this->redirect(
+			'https://my.yoast.com/connect',
+			array(
+				'url'          => WPSEO_Utils::get_home_url(),
+				'client_id'    => $client_id,
+				'extensions'   => array(),
+				'redirect_url' => admin_url( 'admin.php?page=' . WPSEO_Admin::PAGE_IDENTIFIER ),
+			)
+		);
+	}
+
+	/**
+	 * Creates a new MyYoast Client instance.
+	 *
+	 * @codeCoverageIgnore
+	 *
+	 * @return WPSEO_MyYoast_Client Instance of the myyoast client.
+	 */
+	protected function get_client() {
+		static $client;
```

Comment:
Other classes use a protected instance variable for the client. I'd choose consistency and ensure either all classes use statics or all classes use instance variables.